### PR TITLE
Add bash and bats versions to test runner results.json file.

### LINKED ISSUE #557: Pangram test fails with "bad substitution"
The tests for 'pangram' pass locally but fail on the exercism website. I think this is possibly due to different versions of bash and/or bats between my machine and the exercism website. Can anyone direct me to where I might find with version of bash and bats the website is using?

On my machine `bash --version` reports the following: `GNU bash, version 5.1.8(1)-release (x86_64-pc-linux-gnu)`
and `bats --version` reports: `Bats 1.5.0`

Here's the error on the exercism website:
```
(from function 'assert_success' in file bats-extra.bash, line 409,
 in test file pangram.bats, line 11)
  'assert_success' failed

-- command failed --
status : 1
output : pangram.sh: line 34: input=${1@L}: bad substitution
--
```
I get the same error for each of the 10 tests. The line that fails is `local input=${1@L}` in the function `main`.
Here is my code that works on my machine (shellcheck reports nothing):
```bash
#!/usr/bin/env bash

char() {
    printf "%b" "\\$(printf %o "$1")"
}

initialize_alphabet() {
    for((i=97; i < 123; i++)); do
        alphabet["$(char "$i")"]=0
    done
}

record_letters_used() {
    local input=$1
    for((i=0; i < ${#input}; i++)); do
        character="${input:i:1}"
        if [[ "$character" =~ [[:alpha:]] ]]; then
            alphabet["$character"]=1
        fi
    done
}

count_letters_used() {
    declare -i letters=0
    for letter in "${alphabet[@]}"; do
        if ((letter == 1)); then
            ((letters += 1))
        fi
    done
    echo "$letters"
}

main() {
    local input=${1@L}
    local -A alphabet
    initialize_alphabet
    record_letters_used "$input"
    if (("$(count_letters_used)" == 26)); then
        echo "true"
    else
        echo "false"
    fi
}

main "$@"
```

### PR DESCRIPTION
**Pull request by glennj**
This adds a "test-environment" key to the `results.json` file produced by the test runner:
```json
{
  "version": 2,
  "status": "fail",
  "test-environment": {
    "bash": "5.0.17(1)-release",
    "bats": "Bats 1.5.0"
  },
  "tests": [
    {
      "name": "say one",
      "status": "fail",
      "test_code": "run bash first_fails.sh one\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"one\" ]",
      "message": "(in test file this_is_the_test_script.bats, line 6)\n  `[ \"$output\" == \"one\" ]' failed\n"
    },
    {
      "name": "say two",
      "status": "pass",
      "test_code": "run bash first_fails.sh two\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"two\" ]"
    },
    {
      "name": "say three",
      "status": "pass",
      "test_code": "run bash first_fails.sh three\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"three\" ]"
    }
  ]
}
```

Exercism currently doesn't do anything with this, but pending exercism/exercism#6096, it might.

Fixes exercism/bash#557

### CODE REVIEW

File: tests/data/first_fails/expected_results.json
Comment (glennj) [inline tests/data/first_fails/expected_results.json:7]: Due to changes in the `tests.bats` file, this does not _need_ to be in the expected_results. I thought I'd add at least one to show what the actual results will look like.

**Review by kotp — APPROVED** [review]

File: tests/tests.bats
Comment (IsaacG) [inline tests/tests.bats]: Expanding the paths by hand might be easier to read than using a brace expansion.

**Review by IsaacG — APPROVED** [review]

**Review by kotp — APPROVED** [review]

**Comment by glennj** [comment]
@exercism/maintainers-admin I think I need approval from you. My regular bash reviewers have approved.

**Review by kotp — APPROVED** [review]

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**
